Remotion/Data/DomainObjects.UnitTests/DomainObjectTest.cs

```diff
@@ -558,7 +563,25 @@ public void TransactionContext ()
       var transactionContextIndexer = order.TransactionContext;
 
       Assert.That(transactionContextIndexer, Is.InstanceOf(typeof(DomainObjectTransactionContextIndexer)));
-      Assert.That(((DomainObjectTransactionContext)transactionContextIndexer[_transaction]).DomainObject, Is.SameAs(order));
+      Assert.That(transactionContextIndexer[_transaction].DomainObject, Is.SameAs(order));
+    }
+
+    [Test]
+    public void TransactionContext_AfterDeserialization ()
+    {
+      Assert2.IgnoreIfFeatureSerializationIsDisabled();
+
+      var order = _transaction.ExecuteInScope(() => Order.NewObject());
+      var transactionContextIndexer = order.TransactionContext;
+
+      Assert.That(transactionContextIndexer, Is.InstanceOf(typeof(DomainObjectTransactionContextIndexer)));
+      Assert.That(transactionContextIndexer[_transaction].DomainObject, Is.SameAs(order));
+
+      var deserializedOrder = Serializer.SerializeAndDeserialize(order);
+      var newTransactionContextIndexer = deserializedOrder.TransactionContext;
+
+      Assert.That(transactionContextIndexer[_transaction].DomainObject, Is.SameAs(order));
+      Assert.That(transactionContextIndexer[_transaction].DomainObject, Is.Not.SameAs(deserializedOrder));
```

Comment:
```suggestion
      Assert.That(deserializedTransactionContextIndexer [_transaction].DomainObject, Is.SameAs(deserializedOrder));
```
We don't need to re-check the original indexer because it is immutable.